ownership-chain/runtime/laos/src/configs/block_rewards_source.rs

```diff
@@ -1,5 +1,123 @@
-use crate::{Runtime, RuntimeEvent};
+use crate::{Balances, Runtime};
+use frame_support::weights::Weight;
+use pallet_block_rewards_source::{BalanceOf, WeightInfo};
+use pallet_parachain_staking::PayoutReward;
+use sp_runtime::{traits::Zero, DispatchError};
+use sp_std::marker::PhantomData;
 
 impl pallet_block_rewards_source::Config for Runtime {
-	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = pallet_block_rewards_source::weights::SubstrateWeight<Runtime>;
+	type Currency = Balances;
+}
+pub struct BlockRewardsHandlerAdapter<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime: pallet_parachain_staking::Config + pallet_block_rewards_source::Config>
+	PayoutReward<Runtime, BalanceOf<Runtime>> for BlockRewardsHandlerAdapter<Runtime>
+{
+	fn payout_with_computation_cost(
+		_round_index: pallet_parachain_staking::RoundIndex,
+		collator_id: Runtime::AccountId,
+		amount: pallet_block_rewards_source::BalanceOf<Runtime>,
+	) -> Weight {
+		match pallet_block_rewards_source::Pallet::<Runtime>::send_rewards(
+			collator_id,
+			amount.into(),
+		) {
+			Ok(amount) if amount.is_zero() => Weight::zero(),
+			// TODO: In case of a failure in sending rewards, we should return a weight,
+			// since at least one read operation is performed. Additionally, this situation
+			// incurs an extra write operation.
+			_ => <Runtime as pallet_block_rewards_source::Config>::WeightInfo::send_rewards(),
+		}
+	}
+
+	fn payout(
+		delegator_id: &Runtime::AccountId,
+		amount: pallet_block_rewards_source::BalanceOf<Runtime>,
+	) -> Result<pallet_block_rewards_source::BalanceOf<Runtime>, DispatchError> {
+		pallet_block_rewards_source::Pallet::<Runtime>::send_rewards(delegator_id.clone(), amount)
```

Comment:
```suggestion
		pallet_block_rewards_source::Pallet::<Runtime>::send_rewards(delegator_id, amount)
```

Reply:
compiler complains:
```sh
38 |         pallet_block_rewards_handler::Pallet::<Runtime>::send_rewards(destination, amount)
     |         ------------------------------------------------------------- ^^^^^^^^^^^ expected associated type, found `&<Runtime as Config>::AccountId`
     |         |
     |         arguments to this function are incorrect
     |
     = note: expected associated type `_`
                      found reference `&_`
  note: associated function defined here
    --> /home/carla/dev/fv/laos/ownership-chain/pallets/block-rewards-handler/src/lib.rs:65:10
     |
  65 |         pub fn send_rewards(
     |                ^^^^^^^^^^^^
  help: consider using clone here
     |
  38 |         pallet_block_rewards_handler::Pallet::<Runtime>::send_rewards(destination.clone(), amount)
  ```

Reply:
ok, then to match the signatures of two functions, maybe we should pass the `delegator_id` without reference here? in `collator_payout_reward` `collator_id` is `AccountId`

Reply:
I tried and then because it is passed as value I had to clone in other places in the code: for ex:

https://github.com/freeverseio/laos/blob/b55e6900b455687b0346b8179952a787f87bc377/ownership-chain/pallets/parachain-staking/src/lib.rs#L1751-L1752

Reply:
ok, maybe we need to tackle this separately, usually it's always better to pass through reference, so your thinking was right